byterun/alloc.c

```diff
@@ -229,12 +237,17 @@ CAMLprim value caml_update_dummy(value dummy, value newval)
   CAMLassert (tag < No_scan_tag || tag == Double_array_tag);
 
   Tag_val(dummy) = tag;
+#ifdef FLAT_FLOAT_ARRAY
```

Comment:
Same as above.  caml_update_dummy might still be called (for unboxed float record) on a Double_array_tag block, no?